### PR TITLE
Bratseth/instances in deployment xml rebased take 3

### PR DESCRIPTION
**Pull request by bratseth**
Missed one

### CODE REVIEW

**Review by mpolden — APPROVED** [review]

**Comment by bratseth** [comment]
@jonmv clicked the wrong reviewer

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 1]
```diff
@@ -628,8 +627,8 @@ private ActivateResult unexpectedDeployment(ApplicationId application, ZoneId zo
     private LockedApplication withoutDeletedDeployments(LockedApplication application, InstanceName instance) {
         DeploymentSpec deploymentSpec = application.get().deploymentSpec();
         List<Deployment> deploymentsToRemove = application.get().require(instance).productionDeployments().values().stream()
-                                                          .filter(deployment -> ! deploymentSpec.includes(deployment.zone().environment(),
-                                                                                                          Optional.of(deployment.zone().region())))
+                                                          .filter(deployment -> ! deploymentSpec.requireInstance(instance).includes(deployment.zone().environment(),
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: This

**Comment by jonmv** [comment]
Not rotations, but code that is run on prod deployments in any case.

**Comment by jonmv** [comment]
Removed that test ^^

**Pull request merged**